File: Lessons/Lesson 1 - Time Complexity/TapeEquilibrium.cpp
#include <limits>

int solution(vector<int> &A) {
	long sum = 0;
	vector<int>::iterator it;
	for(it = A.begin(); it != A.end(); it++)
		sum += *it;
	long left = 0, right = sum, min = numeric_limits<long>::max();
	long diff;
	for(it = A.begin(); it != A.end()-1; it++) {
		left += *it;
		right -= *it;
		diff = left-right;
		if(diff < 0)
			diff = -diff;
		if(diff < min)
			min = diff;
	}
	return min;
}

File: Lessons/Lesson 1 - Time Complexity/TapeEquilibrium.js

function solution(A) {
    var sum = 0
    var i;
    for(i = 0; i < A.length; i++) {
        sum += A[i];
    }
    var left = 0, right = sum;
    var min = Number.MAX_VALUE;
    var a, diff;
    for(i = 0; i < A.length-1; i++) {
        a = A[i];
        left += a;
        right -= a;
        diff = Math.abs(left-right);
        if(diff < min)
            min = diff;
    }
    return min;
}
